### PR TITLE
Add shop-by-brand page and add ability to Feature brands

### PR DESCRIPTION
**Pull request by drbyte**
`main_page=brands`

Brands marked as "featured" show at the top of the list

Screen Shot 2020-08-11 at 12 31 26 AM

Admin Manufacturers 2020-08-11 at 12 35 24 AM

Admin Manufacturers 2020-08-11 at 12 35 53 AM

### CODE REVIEW

File: zc_install/sql/install/mysql_zencart.sql [hunk 1]
```diff
@@ -885,6 +885,7 @@ CREATE TABLE manufacturers (
   manufacturers_image varchar(255) default NULL,
   date_added datetime default NULL,
   last_modified datetime default NULL,
+  featured tinyint,
```
Comment (lat9) [inline zc_install/sql/install/mysql_zencart.sql:888]: Shouldn't this have a default value?

File: zc_install/sql/updates/mysql_upgrade_zencart_158.sql [hunk 2]
```diff
@@ -36,6 +36,7 @@ TRUNCATE TABLE db_cache;
 
 
 ALTER TABLE layout_boxes ADD plugin_details varchar(100) NOT NULL default '';
+ALTER TABLE manufacturers ADD COLUMN featured tinyint;
```
Comment (lat9) [inline zc_install/sql/updates/mysql_upgrade_zencart_158.sql:39]: Likewise.

File: zc_install/sql/install/mysql_zencart.sql (hunk 1 above)
Comment (drbyte) [reply zc_install/sql/install/mysql_zencart.sql:888]: Yes, while core code won't have any issue with it setting a default will allow custom code that doesn't account for the column to not throw errors.
Thanks.

File: zc_install/sql/updates/mysql_upgrade_zencart_158.sql [hunk 3]
```diff
@@ -36,6 +36,7 @@ TRUNCATE TABLE db_cache;
 
 
 ALTER TABLE layout_boxes ADD plugin_details varchar(100) NOT NULL default '';
+ALTER TABLE manufacturers ADD COLUMN featured tinyint default 0;
```
Comment (lat9) [inline zc_install/sql/updates/mysql_upgrade_zencart_158.sql:39]: Just a question:  Will this cause issues if I backport this to a zc157 installation when the site's upgraded to v158?

Comment (drbyte) [reply zc_install/sql/updates/mysql_upgrade_zencart_158.sql:39]: While that line will fail, zc_install "should" just note it in the `upgrade_exceptions` table and carry on.  Of course, we should test that. I don't recall off the top of my head whether `ADD` vs `ADD COLUMN` is the specific syntax it checks for (or both).

**Comment by scottcwilson** [comment]
Doc update task #3820

**Pull request merged**